parser/V22/parserImpl_V22.go

```diff
@@ -219,7 +219,7 @@ func (p *Parser) parseTrace(trace typesv22.ExecutionTraceV22, mainMsgCid cid.Cid
 		zap.S().Errorf("Error when trying to get block cid from message, txType '%s': %v", txType, err)
 	}
 
-	messageUuid := tools.BuildMessageId(tipsetCid, blockCid, trace.Msg.Cid().String(), parentId)
+	messageUuid := tools.BuildMessageId(tipsetCid, blockCid, trace.Msg.Cid().String(), mainMsgCid.String(), parentId)
```

Comment:
**mainMsgCid** should be placed before **trace.Msg.Cid().String()** to be consistent